docs/content/reference/directives.md

```diff
@@ -22,12 +22,16 @@ enum Role {
 }
 ```
 
-When we next run go generate, gqlgen will add this directive to the DirectiveRoot 
+When we next run go generate, gqlgen will add this directive to the DirectiveRoot
 ```go
 type DirectiveRoot struct {
-	HasRole func(ctx context.Context, next graphql.Resolver, role Role) (res interface{}, err error)
+	HasRole func(ctx context.Context, obj interface{}, next graphql.Resolver, role Role) (res interface{}, err error)
 }
 ```
+*ctx*: passed from the resolver
+*obj*: the payload aggregated so far
+*next*: signals that the directive was successful
```

Comment:
`next` is the middleware pattern, it's not signaling success it's fetching the value by going through the rest of the stack.